pyfar/dsp/filter/audiofilter.py

```diff
@@ -114,16 +114,22 @@ def high_shelve(signal, frequency, gain, order, shelve_type='I',
     order : number
         The shelve order. Must be ``1`` or ``2``.
     shelve_type : str
-        Defines the characteristic frequency. The default is ``'I'``
+        Defines the characteristic frequency. The default is ``'I'``.
 
         ``'I'``
-            defines the characteristic frequency 3 dB below the gain value if
-            the gain is positive and 3 dB above the gain value otherwise
+            Defines the characteristic frequency 3 dB below the `gain` value if
+            the `gain` is positive and 3 dB above the `gain` value if the
+            `gain` is negative.
         ``'II'``
-            defines the characteristic frequency at 3 dB if the gain is
-            positive and at -3 dB if the gain is negative.
+            Defines the characteristic frequency at 3 dB if the `gain` is
+            positive and at -3 dB if the `gain` is negative.
         ``'III'``
-            defines the characteristic frequency at gain/2 dB.
+            Defines the characteristic frequency at `gain`/2 dB.
+
+        For types ``I`` and ``II`` the absolute value of the `gain` must be
+        sufficiently large (>12 dB) to set the characteristic frequency
```

Comment:
Should we be less explicit here? We have not tested it for gains between 6 and 12 dB. We could also write 
```suggestion
        sufficiently large (:math:`\gg3` dB) to set the characteristic frequency
```

Reply:
I tested the error at the characteristic frequency for all gains, it's below 0.5 dB for gains $>\pm 9$ dB. The error does not depend on the characteristic frequency, nor the sampling rate.

![shelve_error](https://user-images.githubusercontent.com/38947363/202906171-f39afda4-79bd-4581-8ab2-57378161000d.png)
```
import pyfar as pf
import numpy as np
import matplotlib.pyplot as plt

fs = 44100
f_c = 1000 # gain at characteristic frequency
order = 2
gains = np.linspace(24, -24, 100)
shelve_types = ['I', 'II']

impulse = pf.signals.impulse(n_samples=fs, sampling_rate=fs)
gain_f_c = np.zeros((len(shelve_types), len(gains))) # gain at f_c
expected_gain_f_c = np.zeros((len(shelve_types), len(gains))) # expected gain
gain_diff = np.zeros((len(shelve_types), len(gains))) # difference from expected gain

expected_gain_f_c[0,:] = gains - np.sign(gains) * 3
expected_gain_f_c[1,:] = np.sign(gains) * 3
# expected_gain_f_c[2,:] = gains / 2

fig, ax = plt.subplots(1,1)
for i, shelve_type in enumerate(shelve_types):
    for j, gain in enumerate(gains):
        ir = pf.dsp.filter.high_shelve(signal=impulse, frequency=f_c, gain=gain, order=order,
                                    shelve_type=shelve_type)
        idx = ir.find_nearest_frequency(f_c)
        gain_f_c[i, j] = 20*np.log10(np.abs(ir.freq[0,idx]))

    gain_diff[i, :] = np.abs(gain_f_c[i, :] - expected_gain_f_c[i, :])
    plt.plot(gains, gain_diff[i,:], label=f'shelve type {shelve_type}', alpha=0.8)

plt.hlines(0.5, np.min(gains), np.max(gains), colors='black', linestyles='dotted')
plt.vlines(-9, 0, 0.5, colors='black', alpha=0.3)
plt.vlines(+9, 0, 0.5, colors='black', alpha=0.3)
plt.xticks((-20, -9, 0, 9, 20))
plt.xlim((np.min(gains), np.max(gains)))
plt.ylim((0, 3))
plt.xlabel('gain in dB')
plt.ylabel('error at characteristic frequency in dB')
plt.legend()
plt.show()
```